src/lib/operators/abstract_join_operator.cpp

```diff
@@ -10,15 +10,15 @@ namespace opossum {
 
 AbstractJoinOperator::AbstractJoinOperator(const std::shared_ptr<const AbstractOperator> left,
                                            const std::shared_ptr<const AbstractOperator> right, const JoinMode mode,
-                                           const std::pair<ColumnID, ColumnID>& column_ids, const ScanType scan_type)
+                                           const JoinColumnIDs& column_ids, const ScanType scan_type)
```

Comment:
ColumnID[2]?

Reply:
We've been using a pair here since forever and I don't see any value in an array.

Reply:
That way we wouldn't have to write `std::pair<ColumnID, ColumnID>` everywhere, which you rightfully don't like because it it is too long, and we wouldn't have to come up with yet another type.

Reply:
The good thing about pair is that they have implicit bounds checking (via `first` and `second`), whereas when I see `_column_id[1]` somewhere I would instantly wonder whether this might go out of bounds. I opt for pairs.